src/api/utils/castutils.py

```diff
@@ -9,3 +9,20 @@ def to_decimal(val, max_digits, precision):
         ctx.prec = max_digits
         places = Decimal(10) ** (precision * -1)
         return Decimal(val).quantize(places, ROUND_HALF_DOWN)
+
+
+def to_number(string):
+    try:
```

Comment:
Also thinking we should protect against bad values here, in case it's ever used by itself. e.g. if `string` was `None` this would throw a `TypeError`, not a `ValueError`. Maybe all these functions should call a string-checking private function that does the same thing?